File: retroauth-android/src/test/java/com/andretietz/retroauth/AndroidAuthenticationHandlerTest.java
package com.andretietz.retroauth;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

import static org.junit.Assert.assertNotNull;
import static org.mockito.Mockito.mock;

@RunWith(RobolectricTestRunner.class)
public class AndroidAuthenticationHandlerTest {

    @Test
    public void createWithTokenTypeFactory() throws Exception {
        Provider provider = mock(Provider.class);
        TokenTypeFactory typeFactory = mock(TokenTypeFactory.class);
        AndroidAuthenticationHandler authenticationHandler = AndroidAuthenticationHandler.Companion
                .create(RuntimeEnvironment.application, provider, typeFactory);
        assertNotNull(authenticationHandler.getMethodCache());
        assertNotNull(authenticationHandler.getOwnerManager());
        assertNotNull(authenticationHandler.getProvider());
        assertNotNull(authenticationHandler.getTokenStorage());
        assertNotNull(authenticationHandler.getTypeFactory());
    }
}

File: retroauth-android/src/test/java/com/andretietz/retroauth/AndroidAuthenticationHandlerTest.kt
package com.andretietz.retroauth

import org.junit.Assert.assertNotNull
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.mock
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment

@RunWith(RobolectricTestRunner::class)
class AndroidAuthenticationHandlerTest {

    @Test
    @Throws(Exception::class)
    fun createWithTokenTypeFactory() {
        val provider = mock(TokenProvider::class.java)
        val typeFactory = mock(TokenTypeFactory::class.java)
        val authenticationHandler = AndroidAuthenticationHandler
                .create(RuntimeEnvironment.application,
                        provider as TokenProvider<AndroidToken>,
                        typeFactory as TokenTypeFactory<AndroidTokenType>)
        assertNotNull(authenticationHandler.methodCache)
        assertNotNull(authenticationHandler.ownerManager)
        assertNotNull(authenticationHandler.provider)
        assertNotNull(authenticationHandler.tokenStorage)
        assertNotNull(authenticationHandler.typeFactory)
    }
}
